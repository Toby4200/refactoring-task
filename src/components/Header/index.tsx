interface Props {
  title: string;
}

const styles = {
  header: {
    color: 'blue',
    boxShadow: '0 0 10px 0 rgba(0, 0, 0, 0.1)',
    padding: '1rem',
  },
  title: {
    fontSize: '2rem',
    fontWeight: 'bold',
  },
};

export default function Header({ title }: Props) {
  return (
    <div style={styles.header}>
      <span style={styles.title}>{title}</span>
    </div>
  );
}
