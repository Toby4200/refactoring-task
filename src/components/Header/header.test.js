import { render, screen } from '@testing-library/react';
import Header from './index';

describe('Header component', () => {
  it('renders title correctly when provided', () => {
    render(<Header title="Posts App" />);
    expect(screen.getByText('Posts App')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Header title="Posts App" />);
    expect(container).toMatchSnapshot();
  });
});
