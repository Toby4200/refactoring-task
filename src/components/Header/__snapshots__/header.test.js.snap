// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<div>
  <div
    style="color: blue; box-shadow: 0 0 10px 0 rgba(0, 0, 0, 0.1); padding: 1rem;"
  >
    <span
      style="font-size: 2rem; font-weight: bold;"
    >
      Posts App
    </span>
  </div>
</div>
`; 